### PR TITLE
fix: empty responses should not be allowed.

### PR DESCRIPTION
**Pull request by kurogeek**
Responses (Replies/Comments) should not be empty.

### CODE REVIEW

**Comment by kurogeek** [comment]
@ssiyad How can I run the check again? I think the error doesn't come from the code. Instead, it came from something else.

**Comment by ssiyad** [comment]
@kurogeek `validate()` does not have to create a toast. It can return an error message which can be handled in `onError`. Can you make the change so that only `onError` create toasts

**Comment by kurogeek** [comment]
@ssiyad got it, done.

File: desk/src/pages/desk/ticket/data.ts [hunk 1]
```diff
@@ -48,6 +49,11 @@ export const useTicketStore = defineStore("ticket", () => {
 				onSuccess() {
 					clean();
 				},
+				validate() {
+					if (isEmpty(editor.content)) {
```
Comment (ssiyad) [inline desk/src/pages/desk/ticket/data.ts]: Use `params`

File: desk/src/pages/desk/ticket/editor/BottomSection.vue [hunk 2]
```diff
@@ -140,6 +142,19 @@ const insertRes = createResource({
...
 	},
+	validate(params) {
+		if (isEmpty(params.doc.content)) {
+			return "Message is empty";
+		}
+	},
+	onError(error) {
+		createToast({
+			title: "Error sending reply",
+			text: error.messages?.join("\n"),
```
Comment (ssiyad) [inline desk/src/pages/desk/ticket/editor/BottomSection.vue:152]: Error from `validate()` will be available as an `Error` object. You can get the message from this with `error.message`

**Review by ssiyad — CHANGES_REQUESTED** [review]
There was a recent addition to editor. A `tiptap` object is added to `editor` which expose all utilities from it. Hence, you can use `editor.tiptap?.isEmpty` instead of `isEmpty` from helpdesk utils

**Comment by kurogeek** [comment]
There is an error `params` is defined but never used. 
Is that okay?

**Comment by ssiyad** [comment]
> There is an error params is defined but never used

You can remove it if not used

File: desk/src/pages/desk/ticket/editor/BottomSection.vue [hunk 3]
```diff
@@ -140,6 +142,19 @@ const insertRes = createResource({
 	onSuccess() {
 		clean();
 	},
+	validate(params) {
+		if (isEmpty(params.doc.content)) {
```
Comment (ssiyad) [inline desk/src/pages/desk/ticket/editor/BottomSection.vue]: We can use `isEmpty` from `tiptap` here too

**Review by ssiyad — CHANGES_REQUESTED** [review]

File: desk/src/pages/desk/ticket/editor/BottomSection.vue (hunk 3 above)
Comment (kurogeek) [reply desk/src/pages/desk/ticket/editor/BottomSection.vue]: I tried, but it didn't work, I was still able to add an empty comment.

Comment (kurogeek) [reply desk/src/pages/desk/ticket/editor/BottomSection.vue]: I used this though `params.doc.tiptap?.isEmpty`

Comment (kurogeek) [reply desk/src/pages/desk/ticket/editor/BottomSection.vue]: or should I just use `editor` object instead of `params`?

Comment (kurogeek) [reply desk/src/pages/desk/ticket/editor/BottomSection.vue]: I just went on using `editor` object anyway.

**Review by ssiyad — APPROVED** [review]
Looks good. Merging

**Pull request merged**